jscomp/core/js_exp_make.ml

```diff
@@ -179,14 +179,20 @@ let make_block ?comment
   (tag_info : J.tag_info) 
   (es : t list) 
   (mutable_flag : J.mutable_flag) : t = 
+  let comment_with_tag = 
+    match comment with 
+    | None -> Lam_compile_util.comment_of_tag_info tag_info 
+    | _ -> comment in
   match tag_info with 
   | Blk_module _ -> 
     {expression_desc = Caml_block(es,mutable_flag, tag,tag_info); comment}
+  | Blk_record des ->
+    let property_map = List.mapi (fun i e -> (des.(i), e)) es in
+    {
+      expression_desc = Object property_map;
+      comment=comment_with_tag 
+    }
   | _ -> 
-  let comment = 
```

Comment:
How about we delay the creation of `Object ..`, stay with `Caml_block`, only convert it to Object in js_dump.ml to preserve existing semantics the most